tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/ContextPropagation.cs

```diff
@@ -32,13 +34,28 @@ public static void InjectTraceIntoRecords<TRecordsRequest>(TRecordsRequest reque
 
             if (request.Records[0].DuckCast<IContainsData>() is { } record)
             {
-                InjectTraceIntoData(record, context);
+                InjectTraceIntoData(record, scope, streamName);
             }
         }
 
-        public static void InjectTraceIntoData<TRecordRequest>(TRecordRequest record, PropagationContext context)
+        public static void InjectTraceIntoData<TRecordRequest>(TRecordRequest record, Scope? scope, string? streamName)
             where TRecordRequest : IContainsData
         {
+            Dictionary<string, object> propagatedContext = new Dictionary<string, object>();
+            if (scope?.Span.Context != null && !string.IsNullOrEmpty(streamName))
+            {
+                var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                if (dataStreamsManager != null && dataStreamsManager.IsEnabled)
+                {
+                    var edgeTags = new[] { "direction:out", $"topic:{streamName}", "type:kinesis" };
```

Comment:
If streamName is nullable, shouldn't we add it only if it has a value ?

Reply:
This is inside an outer block of `if (scope?.Span.Context != null && !string.IsNullOrEmpty(streamName))` so it shouldn't get here if `streamName` has no value.